File: Leetcode/sbgo/Leetcode520-检测大写字母.go
package sbgo

import "unicode"

func detectCapitalUse(word string) bool {
	// 若第 1 个字母为小写，则需额外判断第 2 个字母是否为小写
	if len(word) >= 2 && unicode.IsLower(rune(word[0])) && unicode.IsUpper(rune(word[1])) {
		return false
	}

	// 无论第 1 个字母是否大写，其他字母必须与第 2 个字母的大小写相同
	for i := 2; i < len(word); i++ {
		if unicode.IsLower(rune(word[i])) != unicode.IsLower(rune(word[1])) {
			return false
		}
	}
	return true
}

File: Leetcode/PY/Leetcode520-检测大写字母.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        f = word.split()
        def check(s):
            n,cnt = len(s),0
            for ch in s:
                if ch.isupper(): cnt += 1
            if cnt == 0 or cnt == n: return True
            if cnt == 1 and s[0].isupper(): return True
            return False
        for s in f:
            if not check(s): return False
        return True
